solc/main.cpp

```diff
@@ -53,24 +52,48 @@ static void setDefaultOrCLocale()
 
 int main(int argc, char** argv)
 {
-	setDefaultOrCLocale();
-	solidity::frontend::CommandLineInterface cli(cin, cout, cerr);
-	if (!cli.parseArguments(argc, argv))
+	try
+	{
+		setDefaultOrCLocale();
+		solidity::frontend::CommandLineInterface cli(cin, cout, cerr);
+		if (!cli.parseArguments(argc, argv) || !cli.readInputFiles() || !cli.processInput() || !cli.actOnInput())
+			return 1;
+
+		return 0;
+	}
+	catch (boost::exception const& _exception)
+	{
+		cerr << "Uncaught exception" << boost::diagnostic_information(_exception) << endl;
 		return 1;
-	if (!cli.readInputFiles())
+	}
+	catch (Exception const& _exception)
+	{
+		cerr << "Uncaught exception" << boost::diagnostic_information(_exception) << endl;
 		return 1;
-	if (!cli.processInput())
+	}
+	catch (InternalCompilerError const& _exception)
+	{
+		cerr << "Uncaught exception" << boost::diagnostic_information(_exception) << endl;
 		return 1;
-	bool success = false;
-	try
+	}
+	catch (smtutil::SMTLogicError const& _exception)
 	{
-		success = cli.actOnInput();
+		cerr << "Uncaught exception" << boost::diagnostic_information(_exception) << endl;
+		return 1;
 	}
-	catch (boost::exception const& _exception)
+	catch (std::exception const& _e)
 	{
-		cerr << "Exception during output generation: " << boost::diagnostic_information(_exception) << endl;
-		success = false;
+		cerr() << "Uncaught exception" << (_e.what() ? ": " + string(_e.what()) : ".") << endl;
```

Comment:
I originally did not do this in #10295 but I think that `boost::diagnostic_information` is usable with `std::exception` too. Could you try it?